### ISSUE TITLE
Decide on a CSS library: Tachyons or Tailwind

### ISSUE THREAD

**Issue by tangjeff0** [athensresearch/athens#10]
Both [Tachyons](https://tachyons.io/#style) and [Tailwind](https://tailwindcss.com/docs/utility-first) are more so libraries (i.e. a composable design language) than they are frameworks (e.g. not composable Bootstrap or Bulma). Functional programmers probably like the former more 😄.

**Comment by jeroenvandijk** [comment]
I have no experience on the mentioned libraries (only bootstrap). Composable sounds good :) Do you know examples of websites that were build with the libraries? It could be worth to check how it looks and how the underlying CSS and html is structured.

**Comment by tangjeff0** [comment]
I also haven't used these libraries and haven't looked at examples, so there may be even better options. The ideal would be writing CSS with EDN!

**Comment by tangjeff0** [comment]
@teodorlu Have you used any of these libraries?

**Comment by jeroenvandijk** [comment]
I gave it some thought and I wonder if Tachyons and Tailwind are more modular than something like [Sass](https://sass-lang.com/guide). 

Another usefool tool in this category could be the Clojure library [garden](https://github.com/noprompt/garden). It doesn't give you any styling out of the box, but it might make css easier to work with. 

You could create helper functions that either add certain classes to an element or certain styling. It would not matter much. In these helper functions you could choose to use one of these frameworks.

```
(defn +rounder-corners [attrs]
   (update attrs :class str " tailwind class "))

(defn +rounder-corners [attrs]
   (assoc attrs :style (garden/css {:text-decoration "none"})))

[:div (+rounded-corners {:on-click (fn [_} ...)}) "content here"]
```

If the helpers are given semantic names and organized properly, we might be able to benefit from better reuse as well. Maybe even change themes easily.

**Comment by tangjeff0** [comment]
Themes open up a big can of worms 🤪. Even with better defaults than Roam, users will still want to customize styling, and this is especially true (and encouraged!) for open-source software. Thus, we must also consider what library or framework would be composable and flexible not just for developers, but also for end-users.

Which brings us to the can of worms: isn't styling just one application within a larger plugin architecture? Overriding CSS with a Chrome extension is simply inferior to toggling themes in, say, an IDE. That said, a plugin architecture sounds hard, even one that only handles styling. But if we do go down this route, I bet we can get inspiration from open-source VS Code and Atom :)

**Comment by palashkaria** [comment]
Some general thoughts on this:

Whichever library we choose, the best would be to not let the internal implementation leak. 

There are two things we need to do generally with CSS

1. Layouts
2. Styling

For Layouts, I suggest we have some primitives like `<Stack>` (vertical), `<Columns>` (Horizontal); which have some pre-applied class names/styles & some 'props' to control the spacing between its children. (Internally, it can use flexbox + margins to achieve this.)

This also means that no other components should have any surrounding whitespace

(References: This whole idea comes from [@markdalgeish](https://twitter.com/markdalgleish)'s [braid design system](https://seek-oss.github.io/braid-design-system/) & also 
[@siddharthkp](https://twitter.com/siddharthkp)'s react-ui '[primitives](https://react-ui.dev/components#Primitives)' )



The work on sidebar (#26), for example, can be done with 

```<Stack space="medium">
	<Preview id="id1" /> 
	<Preview id="id2" /> 
	<Preview id="id3" /> 
</Stack>```

For similar reasons, we should think start with an `<EditableBlock>` component, which becomes a `<textarea>` on focus, and parses on focus out. (We might need to check performance issues on this kind of approach; but I think this is what Roam currently does)




PS: Also, as you can see, I am very new to closure, I basically think in JSX! I have started looking in reagent 😅

**Comment by nikhil-varma** [comment]
Tachyons customization can become a pain sometimes especially when theming is into consideration.  It provides a set of variables that you can customize similar to how Bootstrap provides if I am not wrong.... That said, Tachyons might be a slightly better choice over Bootstrap.. 🤔  Also, how impactful would it be to not use a library instead of a fully custom CSS base? 

> Whichever library we choose, the best would be to not let the internal implementation leak.

Agreed.. that may also help migrate over to something else if needed

**Comment by tangjeff0** [comment]
Hi @nikhil-varma, great idea, it's entirely possible we start from the ground up :)

**Comment by Bardia95** [comment]
I have not used Tachyons or Tailwind but I like Tachyons aesthetic better. I will say I have always preferred using plain CSS or SASS (which I would avoid).

**Comment by tangjeff0** [comment]
Currently, I'm wanting to go vanilla. This is more customizable and composable long-term. I don't want to do overrides and dig into Tachyons/Tailwind's CSS source code.

Also, talking with designers, I'm realizing we have very few core components of a design language system. Still want to write CSS in EDN though, with something like Garden :)

**Comment by tangjeff0** [comment]
Resolved by #54. Thanks @Smittyvb !

**Comment by teodorlu** [comment]
> @teodorlu Have you used any of these libraries?

No, sorry; my Clojure experience is mostly JVM. Going with garden for now sounds good to me.
